code/modules/transport/pods/MainWeapon.dm

```diff
@@ -380,6 +380,624 @@
 		opencomputer(usr)
 		return
 
+// Extra-Facility Integrity Fabricator: the big boyo pod construction tool
+
+//mode defines for tool
+#define EFIF_MODE_DRILL 0
+#define EFIF_MODE_FLOORS 1
+#define EFIF_MODE_R_FLOORS 2
+#define EFIF_MODE_WALLS 3
+#define EFIF_MODE_REPAIR 4
+
+//construction time modifier per tile
+#define EFIF_FLOOR_BUILD_TIME 2
+#define EFIF_R_FLOOR_BUILD_TIME 3
+#define EFIF_WALL_BUILD_TIME 6
+//walls will also require the floor build time if built directly on space
+
+//construction cost per tile
+#define EFIF_FLOOR_COST 1
+#define EFIF_R_FLOOR_COST 2
+#define EFIF_WALL_COST 4
+
+//sheet-loading success flags
+#define LOAD_FULL 1
+#define LOAD_SUCCESS 2
+
+TYPEINFO(/obj/item/shipcomponent/mainweapon/constructor)
+	mats = list("metal_superdense" = 50, "claretine" = 20, "electrum" = 10)
+
+/obj/item/shipcomponent/mainweapon/constructor
+	name = "EFIF-1 Construction System"
+	desc = "A pair of elaborate robotic arms equipped for large-scale construction and asteroid demolition."
+	current_projectile = new/datum/projectile/laser/drill/cutter
+	appearanceString = "pod_weapon_efif"
+	icon_state = "constructor"
+	firerate = 12
+	var/mode = EFIF_MODE_REPAIR
+	///Current loaded steel sheets (only accepts steel, as the system's metalforming is designed for it)
+	var/obj/item/sheet/steel_sheets = null
+	///Maximum allowable sheets loaded within tool
+	var/max_sheets = 200
+	///Secondary toggleable setting to increase build size (can be deactivated for fine work)
+	var/wide_field = FALSE
+
+	///Currently active construction fields
+	var/list/active_fields = list()
+
+	attack_self(var/mob/user)
+		if(src.steel_sheets)
+			user.put_in_hand_or_drop(steel_sheets)
+			src.steel_sheets = null
+			boutput(user,SPAN_NOTICE("You eject the steel sheets stored in [src]."))
+			playsound(src, 'sound/items/Deconstruct.ogg', 40, TRUE)
+			return
+		..()
+
+	attackby(obj/item/W, mob/user)
+		var/outcome
+		if (W?.cant_drop)
+			boutput(user, SPAN_ALERT("You can't put that in [src] when it's attached to you!"))
+			return ..()
+		else
+			outcome = sheet_load_helper(W, user)
+			if (outcome & LOAD_SUCCESS)
+				boutput(user, SPAN_NOTICE("You load [W] into [src]."))
+				playsound(src, 'sound/items/Deconstruct.ogg', 40, TRUE)
+			if (outcome & LOAD_FULL)
+				boutput(user, SPAN_NOTICE("[src] is[outcome & LOAD_SUCCESS ? " now" : null] fully loaded."))
+			if (outcome < 1)
+				boutput(user, SPAN_ALERT("[src] only accepts unreinforced steel sheets!"))
+				. = ..()
+
+	//Stopgap discoverability feature, pending pod UI improvement
+	activate()
+		. = ..()
+		if(ship.pilot)
```

Comment:
This causes a runtime on trying to insert the constructor component into a pod

Reply:
screm

Will investigate.

Reply:
Think I have it fixed. Verifying ingame.

Reply:
Yep!